allennlp/tests/training/trainer_test.py

```diff
@@ -678,6 +678,19 @@ def test_restoring_works_with_older_checkpointing(self):
         assert trainer._metric_tracker._best_so_far == 0.1
         assert trainer._metric_tracker._epochs_with_no_improvement == 1
 
+    def test_trainer_can_run_gradient_accumulation(self):
+        trainer = Trainer(model=self.model,
+                          optimizer=self.optimizer,
+                          iterator=self.iterator,
+                          train_dataset=self.instances,
+                          validation_dataset=self.instances,
+                          num_epochs=2,
+                          num_gradient_accumulation_steps=2)
+        assert trainer._num_gradient_accumulation_steps == 2
+        assert trainer._accumulate_gradients
+
+        trainer.train()
```

Comment:
it would really be better if this test was checking that the gradient accumulation was doing the right thing, this is really just testing that it doesn't crash

Reply:
I've added conditions to check the effective number of batches trained.